node/cn/backend.go

```diff
@@ -507,7 +507,13 @@ func (s *CN) ChainDB() database.DBManager        { return s.chainDB }
 func (s *CN) IsListening() bool                  { return true } // Always listening
 func (s *CN) ProtocolVersion() int               { return int(s.protocolManager.SubProtocols[0].Version) }
 func (s *CN) NetVersion() uint64                 { return s.networkId }
-func (s *CN) Downloader() *downloader.Downloader { return s.protocolManager.downloader }
+func (s *CN) Downloader() *downloader.Downloader {
+	dl, ok := s.protocolManager.downloader.(*downloader.Downloader)
+	if !ok {
+		logger.Crit("Failed to covert protocolManager.downlader to *downloader.Downloader")
```

Comment:
Is this an error that causes node shutdown actually?

Reply:
I changed `Downloader()` to `Progress()` to avoid type assertion and simplification. Could you please check?